### PR TITLE
Hub client code review clean up with Tim/Abhipsa

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/DirectMethod/DirectMethodRequest.cs [hunk 1]
```diff
@@ -69,37 +84,28 @@ public DirectMethodRequest()
...
+        /// </summary>
+        [JsonIgnore]
+        protected internal PayloadConvention PayloadConvention { get; set; }
+
         /// <summary>
         /// Method timeout, in seconds.
         /// </summary>
         [JsonProperty("responseTimeoutInSeconds", NullValueHandling = NullValueHandling.Ignore)]
-        protected internal int? ResponseTimeoutInSeconds => ResponseTimeout.HasValue && ResponseTimeout > TimeSpan.Zero
+        internal int? ResponseTimeoutInSeconds => ResponseTimeout.HasValue && ResponseTimeout > TimeSpan.Zero
```
Comment (drwill-ms) [inline iothub/device/src/DirectMethod/DirectMethodRequest.cs:97]: Users will set this via the public properties these read from. This is for serialization only.

File: iothub/device/src/Edge/CustomCertificateValidator.cs [hunk 2]
```diff
@@ -18,59 +17,53 @@ internal class CustomCertificateValidator : ICertificateValidator
...
+            if (Logging.IsEnabled)
+                Logging.Info(this, "CustomCertificateValidator.GetCustomCertificateValidation()", nameof(ICertificateValidator.GetCustomCertificateValidation));
 
             return (sender, cert, chain, sslPolicyErrors) =>
                 ValidateCertificate(_certs.First(), cert, chain, sslPolicyErrors);
         }
 
         private void SetupCertificateValidation()
         {
-            Debug.WriteLine("CustomCertificateValidator.SetupCertificateValidation()");
```
Comment (drwill-ms) [inline iothub/device/src/Edge/CustomCertificateValidator.cs:42]: Use logging instead.

File: iothub/device/src/FileUpload/FileUploadCompletionNotification.cs [hunk 3]
```diff
@@ -10,30 +10,47 @@ namespace Microsoft.Azure.Devices.Client.Transport
     /// </summary>
     public class FileUploadCompletionNotification
     {
+        /// <summary>
+        /// Used to tell the service that the client is done uploading a file with the specified correlation Id.
+        /// </summary>
+        /// <param name="correlationId">The correlation Id of the SAS URI.</param>
+        /// <param name="isSuccess">Whether the file upload was successful or not.</param>
+        public FileUploadCompletionNotification(string correlationId, bool isSuccess)
```
Comment (drwill-ms) [inline iothub/device/src/FileUpload/FileUploadCompletionNotification.cs:18]: Make required properties required in the ctor, and readonly.

File: iothub/device/src/IotHubBaseClient.cs [hunk 4]
```diff
@@ -451,49 +451,42 @@ internal async Task OnMethodCalledAsync(DirectMethodRequest directMethodRequest)
...
+
+                directMethodResponse.RequestId = directMethodRequest.RequestId;
+                directMethodResponse.PayloadConvention = _clientOptions.PayloadConvention;
 
-            if (_deviceDefaultMethodCallback == null)
+                await SendDirectMethodResponseAsync(directMethodResponse).ConfigureAwait(false);
+            }
+            catch (Exception ex)
             {
-                directMethodResponse = new DirectMethodResponse((int)DirectMethodResponseStatusCode.MethodNotImplemented)
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:463]: Don't respond to direct methods on behalf of the user as we don't know how they interpret HTTP status codes. Just log it and don't respond to the request.

File: iothub/device/src/IotHubModuleClient.cs [hunk 5]
```diff
@@ -265,16 +265,9 @@ private async Task<DirectMethodResponse> InvokeMethodAsync(Uri uri, DirectMethod
                 };
 
                 using var httpTransport = new HttpTransportHandler(pipelineContext, transportSettings, httpClientHandler);
-                var methodInvokeRequest = new DirectMethodRequest
```
Comment (drwill-ms) [inline iothub/device/src/IotHubModuleClient.cs:268]: No need to clone.

File: iothub/device/src/Edge/CustomCertificateValidator.cs [hunk 6]
```diff
@@ -18,59 +17,53 @@ internal class CustomCertificateValidator : ICertificateValidator
...
-                    mqttTransportSettings.RemoteCertificateValidationCallback =
-                        (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
-                }
+                mqttTransportSettings.RemoteCertificateValidationCallback ??=
+                    (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
             }
-
-            // TODO: Tim/Abhipsa, what does this mean?
-
-            // InvokeMethodAsync is over HTTP even when transportSettings set a different protocol
```
Comment (abhipsaMisra) [inline iothub/device/src/Edge/CustomCertificateValidator.cs:63]: For reference, the cert validator for invoke methods over http is set here: https://github.com/Azure/azure-iot-sdk-csharp/blob/672b447887df5faf0018d279d4508462a761c388/iothub/device/src/IotHubModuleClient.cs#L257

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**